### ISSUE TITLE
Account for incorrect `[].into_iter()` in <2021 edition

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#85189]
Currently uses of `<[_; _] as _>::into_iter()` [emit a future compat warning](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f2b5696e9655f77ef4438554c9f892ec):

```
warning: this method call currently resolves to `<&[T; N] as IntoIterator>::into_iter` (due to autoref coercions), but that might change in the future when `IntoIterator` impls for arrays are added.
 --> src/main.rs:2:23
  |
2 |     let _ = [1, 2, 3].into_iter();
  |                       ^^^^^^^^^ help: use `.iter()` instead of `.into_iter()` to avoid ambiguity: `iter`
  |
  = note: `#[warn(array_into_iter)]` on by default
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #66145 <https://github.com/rust-lang/rust/issues/66145>
```

but at some point this will cause ambiguity errors on 2015 and 2018 editions. On 2021 onwards, this case is handled by a special attribute and keeps the code compiling.

We should detect that case explicitly and provide guidance about what to write in the current edition, or to update to the new edition.

**Comment by m-ou-se** [comment]
That code now produces:
```
warning: this method call resolves to `<&[T; N] as IntoIterator>::into_iter` (due to backwards compatibility), but will resolve to <[T; N] as IntoIterator>::into_iter in Rust 2021.
 --> src/main.rs:2:23
  |
2 |     let _ = [1, 2, 3].into_iter();
  |                       ^^^^^^^^^
  |
  = note: `#[warn(array_into_iter)]` on by default
  = warning: this changes meaning in Rust 2021
  = note: for more information, see issue #66145 <https://github.com/rust-lang/rust/issues/66145>
help: use `.iter()` instead of `.into_iter()` to avoid ambiguity
  |
2 |     let _ = [1, 2, 3].iter();
  |                       ^^^^
help: or use `IntoIterator::into_iter(..)` instead of `.into_iter()` to explicitly iterate by value
  |
2 |     let _ = IntoIterator::into_iter([1, 2, 3]);
  |             ^^^^^^^^^^^^^^^^^^^^^^^^         ^
```
